accounts-db/src/accounts_db.rs

```diff
@@ -3506,6 +3508,31 @@ impl AccountsDb {
         let alive_accounts_collect = Mutex::new(T::with_capacity(len, slot));
         let pubkeys_to_unref_collect = Mutex::new(Vec::with_capacity(len));
         let zero_lamport_single_ref_pubkeys_collect = Mutex::new(Vec::with_capacity(len));
+        let mut obsolete_accounts_filtered = 0;
+
+        // Get a set of all obsolete offsets
+        // Slot is not needed, as all obsolete accounts can be considered
+        // dead for shrink. Zero lamport accounts are not marked obsolete
+        let obsolete_offsets: AHashSet<_> = store
+            .get_obsolete_accounts(None)
+            .iter()
+            .map(|(offset, _)| *offset)
```

Comment:
nit: looks like `data_len` is not needed. However, we still collect them in a vec and then discard the vec. seems like we can save vec allocation. if we move this method on `AccountStorageEntry`, i.e. `get_obsolete_accounts_offsets(slot)`.

Reply:
what's the plan to address this? 

I see that @brooksprumo had a similar comments for this. 
https://github.com/anza-xyz/agave/pull/6632/files#r2162529131

Are we decided to do return the iterator in future PR instead of the suggestion above?

Reply:
ok.